### PR TITLE
Nameable Banner API

### PR DESCRIPTION
**Pull request by Owen1212055**
Supercedes https://github.com/PaperMC/Paper/pull/5867

### CODE REVIEW

File: patches/server/0890-Nameable-Banner-API.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
++
++    @Override
++    public String getCustomName() {
++        return io.papermc.paper.adventure.PaperAdventure.LEGACY_SECTION_UXRC.serializeOrNull(this.customName());
++    }
++
++    @Override
++    public void setCustomName(String name) {
++       this.customName(io.papermc.paper.adventure.PaperAdventure.LEGACY_SECTION_UXRC.deserializeOrNull(name));
++    }
```
Comment (Machine-Maker) [inline patches/server/0890-Nameable-Banner-API.patch]: How do we feel about adding new deprecated methods? I know its on the interface, but is there a better way to do this? Like should we make those methods on the Nameable interface be defaults that delegate to the component methods with the right serializer?

Comment (Owen1212055) [reply patches/server/0890-Nameable-Banner-API.patch]: I mean id say we can just make these default methods in the adventure patch yeah.

**Comment by Owen1212055** [comment]
Unstale

**Review by Machine-Maker — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

**Comment by Owen1212055** [comment]
Rebased, tested and updated

**Pull request merged**